nautobot_circuit_maintenance/handle_notifications/sources.py

```diff
@@ -516,6 +529,54 @@ def receive_notifications(self, logger: Job, since: int = None) -> Iterable[Main
         return received_notifications
 
 
+class RedirectAuthorize(Exception):
+    """Custom class to signal a redirect to trigger OAuth autorization workflow for a specific source_slug."""
+
+    def __init__(self, url_name, source_slug):
+        """Init for RedirectAuthorize."""
+        self.url_name = url_name
+        self.source_slug = source_slug
+        super().__init__()
+
+
+class GmailAPIOauth(GmailAPI):
+    """GmailAPIOauth class.
+
+    See: https://developers.google.com/identity/protocols/oauth2/web-server
+    """
+
+    def load_credentials(self):
+        """Load Gmail API OAuth credentials."""
+        notification_source = NotificationSource.objects.get(name=self.name)
+        try:
+            self.credentials = notification_source.get_token()
+        except EOFError:
+            # _token has not been initialized yet
```

Comment:
Should we log a message here? Not sure if that should be a debug message or a warning message though?